src/timeMachine/components/SubmitQueryButton.tsx

```diff
@@ -77,6 +77,10 @@ class SubmitQueryButton extends PureComponent<Props> {
   }
 
   componentWillUnmount() {
+    // Clearing and removing timer before the component is unmounted
+    clearTimeout(this.timer)
+    delete this.timer
```

Comment:
why do we want to delete the timer before the component unmounts?

Reply:
because `setTimeout` tries to delete it after the component has been unmounted. So, deleting the timer here stops setTimeout from happening.

Reply:
right here: https://github.com/influxdata/ui/pull/1922/files/588effe69d43f55b975af95a4d4a109328e43ff6#diff-6504aaf0fbb0dc6a4a0968ece1d85325956fa677fde7ce63d5ebb35ee0c36360R62